packages/legend-query/src/components/QueryBuilderPropertySearchPanel.tsx

```diff
@@ -0,0 +1,641 @@
+                    <div
+                      className={clsx(
+                        'query-builder-property-search-panel__form__section__toggler',
+                        {
+                          'query-builder-property-search-panel__form__section__toggler--disabled':
+                            false,
+                        },
+                      )}
+                    >
+                      <button
+                        className={clsx(
+                          'query-builder-property-search-panel__form__section__toggler__btn',
+                          {
+                            'query-builder-property-search-panel__form__section__toggler__btn--toggled':
+                              propertySearchPanelState.isOneManyRowsIncluded,
+                          },
+                        )}
+                        onClick={toggleOneManyRowsIncluded}
+                        tabIndex={-1}
+                      >
+                        {propertySearchPanelState.isOneManyRowsIncluded ? (
+                          <CheckSquareIcon />
+                        ) : (
+                          <SquareIcon />
+                        )}
+                      </button>
+                      <div className="query-builder-property-search-panel__form__section__toggler__prompt">
+                        Included
+                      </div>
+                    </div>
+                  </div>
+                  <div className="query-builder-property-search-panel__form__section">
+                    <div className="query-builder-property-search-panel__form__section__header__label">
+                      Element types
+                    </div>
+                    <div className="query-builder-property-search-panel__filter__element">
+                      <button
+                        className={clsx(
+                          'query-builder-property-search-panel__form__section__toggler__btn',
+                          {
+                            'query-builder-property-search-panel__form__section__toggler__btn--toggled':
+                              propertySearchPanelState.isClassIncluded,
+                          },
+                        )}
+                        onClick={toggleClassIncluded}
+                        tabIndex={-1}
+                      >
+                        {propertySearchPanelState.isClassIncluded ? (
+                          <CheckSquareIcon />
+                        ) : (
+                          <SquareIcon />
+                        )}
+                      </button>
+                      <div className="query-builder-property-search-panel__form__section__toggler__prompt">
+                        Class
+                      </div>
+                      <div
+                        className="query-builder-property-search-panel__only__element"
+                        onClick={setOnlyClassIncluded}
+                      >
+                        only
```

Comment:
I think this is a niche feature that we don't need for now. I would be more comfortable not having this implemented in the first round to reduce code complexity. If this proves to be a highly desirable feature, let's add it then. Sorry about this !